packages/plugins/src/plugins/diagnostic-comment.plugin.ts

```diff
@@ -0,0 +1,238 @@
+import { createRequire } from 'node:module';
+import { DiagnosticWithContext, hints } from '@rehearsal/codefixes';
+import {
+  Plugin,
+  PluginOptions,
+  type PluginResult,
+  PluginsRunnerContext,
+  Service,
+} from '@rehearsal/service';
+import { findNodeAtPosition } from '@rehearsal/ts-utils';
+import debug from 'debug';
+import ts, { LanguageService } from 'typescript';
+import {
+  findDiagnosticNode,
+  getConditionalCommentPos,
+  inJsxText,
+  inTemplateExpressionText,
+  onMultilineConditionalTokenLine,
+} from './utils.js';
+import type MS from 'magic-string';
+
+const require = createRequire(import.meta.url);
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+const MagicString = require('magic-string');
```

Comment:
why the require? magic-string exports as ESM `import MagicString from 'magic-string';`

Reply:
I have tried all combinations of this and it just doesn't work. When you do `import MagicString from 'magic-string';` and you try and call it you get the following type errors.

<img width="1070" alt="Screenshot 2023-04-14 at 10 24 19 AM" src="https://user-images.githubusercontent.com/183799/232072189-47ea3275-2fc1-462a-87f8-c589a07abaa8.png">

When you invoke it as `new MagicString.default` you don't get type errors.

<img width="538" alt="Screenshot 2023-04-14 at 10 24 33 AM" src="https://user-images.githubusercontent.com/183799/232072343-bf627776-9df5-4e43-b031-1e7e16c166ab.png">

However you get runtime errors

<img width="917" alt="Screenshot 2023-04-14 at 10 24 41 AM" src="https://user-images.githubusercontent.com/183799/232072411-7b95429b-1c87-4672-9bfa-c92259f39846.png">